packages/material-ui-lab/src/DateRangePicker/DateRangePicker.test.tsx

```diff
@@ -59,15 +59,31 @@ describe('<DateRangePicker />', () => {
     expect(getAllByMuiTest('DateRangeHighlight')).to.have.length(24);
   });
 
+  it('allows a single day range', () => {
+    render(
+      <DesktopDateRangePicker
+        renderInput={defaultRangeRenderInput}
+        onChange={() => {}}
+        value={[
+          adapterToUse.date('2018-01-01T00:00:00.000'),
+          adapterToUse.date('2018-01-01T00:00:00.000'),
+        ]}
+      />,
```

Comment:
I was sceptic that we should allow this by default but then realized it is behind a flag (`allowSameDateSelection` prop).

However, the test makes it seem like either it is default behavior or `allowSameDateSelection` is not working.

If you expect a range as an input that you usually don't expect a single date. I've worked on some internal tools for car-rentals in the past and they had the notion of "minimum range" which would make more sense here. Though it looks like DateRangePicker doesn't implement such a feature?

Reply:
I believe:

- `allowSameDateSelection` is leaking from another type, it shouldn't end up in the public API
- we miss the capability to customize the valid ranges, we started to discuss this in https://github.com/mui-org/material-ui-pickers/issues/1759#issuecomment-626604129

Reply:
Might be possible with existing API. I haven't looked so far. Still scratching the surface and ironing out kinks from the forceful integration.